lib/logstash/inputs/lumberjack.rb

```diff
@@ -43,10 +43,10 @@ def register
 
   public
   def run(output_queue)
-    @lumberjack.run do |l|
-      @codec.decode(l.delete("line")) do |event|
+    @lumberjack.run(@codec.clone) do |lines, codec|
```

Comment:
this has a bug because @codec.clone is only invoked once, so each thread shares that clone.

We can avoid monkeypatching Lumberjack::Server (as mentioned in a comment below) and use `#accept` in a loop, dispatching threads ourselves:

```ruby
while true # or not shutting down?
  connection = @lumberjack.accept # blocks until accept
  Thread.new(connection, @codec.clone) do |connection, codec|
    connection.run do |line|
      codec.decode(line.delete("line")) do |event|
        ...
      end
    end
  end
end
```

(Code above should probably not just be in one method, but split out for easier testing.)